reporter.cpp

```diff
@@ -19,6 +20,11 @@ __attribute__((constructor)) static void check_silence() {
   SILENT_REPORTER = (std::getenv("SILENT_REPORTER") != nullptr);
 }
 
+static int MAX_REPORT_SIZE = 10;
+__attribute__((constructor)) static void check_max_report() {
+  MAX_REPORT_SIZE = (std::getenv("MAX_REPORT_SIZE") != nullptr);
```

Comment:
Why don't you `atoi(std::genenv())` so you can manually set max report size using env var.